src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,239 @@
+    }
+
+    /**
+     * Creates a formatted string to store the end date and time.
+     *
+     * @return The formatted string to store the end date and time.
+     */
+    public String getEventToForSave() {
+        return this.toDate == null
+                ? (this.toDateTime.getYear() + "-" + padZero(this.toDateTime.getMonthValue())
+                + "-" + padZero(this.toDateTime.getDayOfMonth()) + " " + padZero(this.toDateTime.getHour()) + ":"
+                + padZero(this.toDateTime.getMinute()))
+                : (this.toDate.getYear() + "-" + padZero(this.toDate.getMonthValue())
+                + "-" + padZero(this.toDate.getDayOfMonth()));
+    }
+
+    /**
+     * Creates a formatted string to display the start date and time.
+     *
+     * @return The formatted string to display the start date and time.
+     */
+    public String getEventFromForDisplay() {
+        return this.fromDate == null
+                ? (this.fromDateTime.getMonth() + " " + padZero(this.fromDateTime.getDayOfMonth())
+                + " " + this.fromDateTime.getYear() + " " + padZero((this.fromDateTime.getHour() > 12
+                ? this.fromDateTime.getHour() - 12 : this.fromDateTime.getHour())) + ":"
+                + padZero(this.fromDateTime.getMinute()) + (this.fromDateTime.getHour() >= 12 ? "PM" : "AM"))
+                : (this.fromDate.getMonth() + " " + padZero(this.fromDate.getDayOfMonth())
+                + " " + this.fromDate.getYear());
+    }
+
+    /**
+     * Creates a formatted string to display the end date and time.
+     *
+     * @return The formatted string to display the end date and time.
+     */
+    public String getEventToForDisplay() {
+        return this.toDate == null
+                ? (this.toDateTime.getMonth() + " " + padZero(this.toDateTime.getDayOfMonth())
+                + " " + this.toDateTime.getYear() + " " + padZero((this.toDateTime.getHour() > 12
+                ? this.toDateTime.getHour() - 12 : this.toDateTime.getHour())) + ":"
+                + padZero(this.toDateTime.getMinute()) + (this.toDateTime.getHour() >= 12 ? "PM" : "AM"))
+                : (this.toDate.getMonth() + " " + padZero(this.toDate.getDayOfMonth())
+                + " " + this.toDate.getYear());
+    }
+
+    /**
+     * Returns the formatted string containing done status and description of the task.
+     *
+     * @return The formatted string containing done status and description of the task.
+     */
+    @Override
+    public String getDescriptionStatus() {
+        return "[E]" + (isDone ? "[X] " : "[ ] ") + this.description + " (from: " + this.getEventFromForDisplay()
+                + " to: " + this.getEventToForDisplay() + ")";
+    }
+
+    /**
+     * Returns various fields of the task.
+     *
+     * @return A string array containing the description, done status, and formatted due date and time of a task.
+     */
+    @Override
+    public String[] getFields() {
+        String[] result = new String[4];
```

Comment:
Perhaps the variable `result` can be in plural?
> Plural form should be used on names representing a collection of objects. &ndash; se-education.org

Here is an example:
```java
Collection<Point> points;
int[] values;
```
Refer for the documentation [here](https://se-education.org/guides/conventions/java/intermediate.html).

Reply:
Ah true!